src/mixtures/mixturemodel.jl

```diff
@@ -135,13 +135,13 @@ function MixtureModel(components::Vector{C}, prior::Categorical) where C<:Distri
     MixtureModel{VF,VS,C}(components, prior)
 end
 
-MixtureModel(components::Vector{C}, p::Vector{Float64}) where {C<:Distribution} =
+MixtureModel(components::Vector{C}, p::Vector{T}) where {C<:Distribution,T<:Real} =
```

Comment:
Should we allow for any `<:AbstractVector{<:Real}`?

Reply:
Is this the only place we need to change? Or we actually need make `AT<: AbstractVector` part of the parameter of type `MixtureModel`?

Reply:
yes exactly